### DISCUSSION TITLE
Use generator after instance creation ?

### DISCUSSION THREAD

**Discussion by nicolasvac** [Minestom/Minestom#1073] [Help]
Hello,

in my code i'm creating an istance for when the user joins, setting its spawn point,

then i wait for PlayerSpawnEvent to come and then i would like to use Generator to generate the world so the player can see the
world generating real time.

How can i trigger the world generation after the instance has already been created ?

Thanks

**Comment by DasLixou** [comment]
You declare the generator after the `instanceManager.createInstanceContainer()`. When the player spawns at this point, the instance will load the required chunks with the generator. At creation of the instance, no chunk will generate before its required.

**Comment by DasLixou** [comment]
Ah yes. When you want to see it real-time, first you have to load the chunk the player spawns manually. But Minestom will by itself load it and all the others around it and send it to the player.
